src/lib.rs

```diff
@@ -1032,79 +1032,156 @@ pub struct Keys<'a, K: 'a, V: 'a> {
 impl<'a, K, V> Iterator for Keys<'a, K, V> {
     type Item = &'a K;
 
+    #[inline]
     fn next(&mut self) -> Option<&'a K> {
         self.iter.next().map(|ent| &ent.key)
     }
 
+    #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
         self.iter.size_hint()
     }
+
+    #[inline]
+    fn count(self) -> usize {
+        self.len()
+    }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        self.iter.nth(n).map(|ent| &ent.key)
+    }
+
+    #[inline]
+    fn last(mut self) -> Option<Self::Item> {
+        self.next_back()
+    }
 }
 
 impl<'a, K, V> DoubleEndedIterator for Keys<'a, K, V> {
+    #[inline]
     fn next_back(&mut self) -> Option<&'a K> {
         self.iter.next_back().map(|ent| &ent.key)
     }
 }
 
+impl<'a, K, V> ExactSizeIterator for Keys<'a, K, V> {}
```

Comment:
ESI is intentionally left out, because it's not really compatible with tombstones if that's the approach that will be adopted later.